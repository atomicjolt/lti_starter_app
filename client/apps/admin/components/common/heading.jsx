import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { Wrapper, Button, Menu, MenuItem } from 'react-aria-menubutton';
import { Link } from 'react-router';
import assets from '../../libs/assets';
import SubNav from '../common/sub_nav';

const select = state => ({
  userName: state.settings.display_name,
  signOutUrl: state.settings.sign_out_url,
  sites: state.sites,
});

function handleSelection(value) {
  window.location = value;
}

export function Heading(props) {
  const img = assets('./images/atomicjolt.svg');

  let back = null;
  if (props.backTo) {
    back = (
      <button className="c-btn c-btn--back">
        <Link to={props.backTo}>
          <i className="i-back" />
          Back
        </Link>
      </button>
    );
  }

  return (
    <div>
      <header className="c-head">
        <div className="c-back">
          {back}
        </div>
        <img className="c-logo" src={img} alt="Atomic Jolt Logo" />
        <Wrapper
          className="c-user"
          onSelection={handleSelection}
        >
          <Button className="c-username">
            <span>{props.userName}</span>
            <i className="i-dropdown" />
          </Button>
          <Menu className="c-dropdown">
            <ul>
              <li>
                <MenuItem
                  value={props.signOutUrl}
                  text="Logout"
                  className="c-menu-item"
                >
                  <a href={props.signOutUrl}><span>Logout</span></a>
                </MenuItem>
              </li>
              <li>
                <MenuItem
                  value={`${props.signOutUrl}?destroy_authentications=true`}
                  text="Delete"
                  className="c-menu-item"
                >
                  <a href={`${props.signOutUrl}?destroy_authentications=true`}>
                    <span>Delete Canvas Authentications and Sign Out</span>
                  </a>
                </MenuItem>
              </li>
            </ul>
          </Menu>
        </Wrapper>
      </header>
      <SubNav sites={props.sites} />
    </div>
  );
}

Heading.propTypes = {
  backTo: PropTypes.string,
  userName: PropTypes.string,
  signOutUrl: PropTypes.string.isRequired,
  sites: PropTypes.shape({}).isRequired,
};


export default connect(select)(Heading);
