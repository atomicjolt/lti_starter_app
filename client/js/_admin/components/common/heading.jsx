import React                               from 'react';
import { connect }                         from 'react-redux';
import { Wrapper, Button, Menu, MenuItem } from 'react-aria-menubutton';
import { Link }                            from 'react-router';
import assets                              from '../../../libs/assets';
import SubNav                              from '../common/sub_nav';

const select = state => ({
  userName: state.settings.display_name,
  signOutUrl: state.settings.sign_out_url,
});

function handleSelection(value) {
  window.location = value;
}

export function Heading(props) {
  const img = assets('./images/atomicjolt.svg');

  let back = null;
  if (props.backTo) {
    back = (
      <button className="c-btn c-btn--back">
        <Link to={props.backTo}>
          <i className="i-back" />
          Back
        </Link>
      </button>
    );
  }

  return (
    <div>
      <header className="c-head">
        <div className="c-back">
          {back}
        </div>
        <img className="c-logo" src={img} alt="Atomic Jolt Logo" />
        <Wrapper
          className="c-user"
          onSelection={handleSelection}
        >
          <Button className="c-username">
            <span>{props.userName}</span>
            <i className="i-dropdown" />
          </Button>
          <Menu className="c-dropdown">
            <ul>
              <li>
                <MenuItem
                  value={props.signOutUrl}
                  text="Logout"
                  className="c-menu-item"
                >
                  <a href={props.signOutUrl}><span>Logout</span></a>
                </MenuItem>
              </li>
              <li>
                <MenuItem
                  value={`${props.signOutUrl}?destroy_authentications=true`}
                  text="Delete"
                  className="c-menu-item"
                >
                  <a href={`${props.signOutUrl}?destroy_authentications=true`}>
                    <span>Delete Canvas Authentications and Sign Out</span>
                  </a>
                </MenuItem>
              </li>
            </ul>
          </Menu>
        </Wrapper>
      </header>
      <SubNav />
    </div>
  );
}

Heading.propTypes = {
  backTo: React.PropTypes.string,
  userName: React.PropTypes.string,
  signOutUrl: React.PropTypes.string.isRequired,
};


export default connect(select)(Heading);
