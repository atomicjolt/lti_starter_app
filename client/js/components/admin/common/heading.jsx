import React        from 'react';
import assets       from '../../../libs/assets';

export default function heading(props) {
  const img = assets('./images/atomicjolt.svg');

  return (
    <header className="c-head">
      <div className="c-back">
        {
          props.back ? <button className="c-btn c-btn--back" onClick={props.back}>
            <i className="i-back" />
            Back
          </button> : null
        }
      </div>
      <img className="c-logo" src={img} alt="Atomic Jolt Logo" />
      <ul className="c-user">
        <li>
          <div className="c-username">{props.userName}<i className="i-dropdown" /></div>
        </li>
      </ul>
    </header>
  );
}

heading.propTypes = {
  back: React.PropTypes.func,
  userName: React.PropTypes.string,
};
