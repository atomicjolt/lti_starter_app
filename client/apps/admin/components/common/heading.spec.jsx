import React from 'react';
import { shallow } from 'enzyme';
import { Heading } from './heading';

jest.mock('../../libs/assets');
describe('common heading', () => {
  let result;
  let props;
  const sites = { 1: { id: 1, oauth_key: 'akey', oauth_secret: 'secret' } };

  beforeEach(() => {
    props = {
      backTo: '/',
      userName: 'username',
      signOutUrl: 'https://www.example.com',
      sites,
    };
    result = shallow(<Heading {...props} />);
  });

  it('matches the snapshot', () => {
    expect(result).toMatchSnapshot();
  });
});
