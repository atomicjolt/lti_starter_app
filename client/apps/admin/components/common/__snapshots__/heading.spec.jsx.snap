// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`common heading matches the snapshot 1`] = `
<div>
  <header
    className="c-head"
  >
    <div
      className="aj-flex"
    >
      <img
        alt="Atomic Jolt Logo"
        className="c-head-img"
      />
      <nav
        className="c-head-nav"
      >
        <Link
          aria-selected={null}
          className="c-head-link"
          key="Atomic Search0"
          onlyActiveOnIndex={false}
          style={Object {}}
        >
          Search
        </Link>
        <Link
          aria-selected={null}
          className="c-head-link"
          key="Atomic Curriculum Tools1"
          onlyActiveOnIndex={false}
          style={Object {}}
        >
          Act
        </Link>
        <Link
          aria-selected={null}
          className="c-head-link"
          key="2"
          onlyActiveOnIndex={false}
          style={Object {}}
        >
          Assessments
        </Link>
        <Link
          aria-selected={null}
          className="c-head-link"
          key="3"
          onlyActiveOnIndex={false}
          style={Object {}}
        >
          Polls
        </Link>
        <Link
          aria-selected={null}
          className="c-head-link"
          key="4"
          onlyActiveOnIndex={false}
          style={Object {}}
        >
          Discussions
        </Link>
        <Link
          aria-selected={null}
          className="c-head-link"
          key="5"
          onlyActiveOnIndex={false}
          style={Object {}}
        >
          Journals
        </Link>
      </nav>
    </div>
    <div
      className="c-head-profile"
    >
      <Menu>
        <Component />
      </Menu>
    </div>
  </header>
  <div
    className="c-yellow-bar"
  />
  <SubNav
    sites={
      Object {
        "1": Object {
          "id": 1,
          "oauth_key": "akey",
          "oauth_secret": "secret",
        },
      }
    }
  />
</div>
`;
